replica/table.cc

```diff
@@ -2787,7 +2788,9 @@ class compaction_group::table_state : public compaction::table_state {
         return _t.make_sstable();
     }
     sstables::sstable_writer_config configure_writer(sstring origin) const override {
-        return _t.get_sstables_manager().configure_writer(std::move(origin));
+        auto cfg = _t.get_sstables_manager().configure_writer(std::move(origin));
+        cfg.erm = _t.get_effective_replication_map();
```

Comment:
It's enough to write [first, last] (or indeed, drop the sharding metadata section for sstables owned by tablets, since with compaction groups they will have no holes).

Reply:
I guess it generates a circular dependency, since compaction groups depend on fixing up sharding.